torch/csrc/generic/utils.h

```diff
@@ -11,7 +11,7 @@
 struct THPStorage;
 struct THSPTensor;
 
-typedef class THPPointer<THWStorage>      THWStoragePtr;
+typedef class THPPointer<c10::StorageImpl>      THStorageImplPtr;
```

Comment:
I'm not sure if `THStorageImplPtr` is the best name for this, so I'm open to other suggestions

Reply:
Come to think of it, do you need to wrap c10::StorageImpl in THPPointer? StorageImpl is refcounting itself.

Reply:
Yes, since you're changing the name already might as well use `at::Storage` or `c10::intrusive_ptr<c10::StorageImpl>`.